src/Transport/TestTransport.php

```diff
@@ -263,6 +284,10 @@ public function send($what): Envelope
 
         $envelope = Envelope::wrap($what);
 
+        if ($this->clock && $delayStamp = $envelope->last(DelayStamp::class)) {
+            $envelope = $envelope->with(AvailableAtStamp::fromDelayStamp($delayStamp, $this->clock->now()));
```

Comment:
I decided to create a new stamp here. Problem is that delay stamp only gives a "delay period", and we want an "availableAt" info.

Another solution would have been to do the same thing than what is done in `InMemoryTransport` and to affect an id to a message with a stamp, and store inside the transport the `availableAt` info, but I tend to prefer my approach

Reply:
I like this stamp too. Let's roll with it.